src/Request.php

```diff
@@ -111,9 +111,19 @@ public function hasHeader($name)
         return !!$this->getHeader($name);
     }
 
+    /**
+     * Checks if the request headers contain the `Expect: 100-continue` header.
+     *
+     * This header MAY be included when an HTTP/1.1 client wants to send a bigger
+     * request body.
+     * See [`writeContinue()`] for more details.
+     *
+     * @return bool
+     * @see Response::writeContinue()
+     */
     public function expectsContinue()
     {
-        return isset($this->headers['Expect']) && '100-continue' === $this->headers['Expect'];
+        return '100-continue' === $this->getHeaderLine('Expect');
```

Comment:
This should also check the protocol version, as it must return `false` for HTTP/1.0.

> A server that receives a 100-continue expectation in an HTTP/1.0 request MUST ignore that expectation.